woocommerce/woocommerce-android
# Multiple coupon support in Order creation

## Multiple coupon support in Order creation


Closes: #9276 


### Description

The flow of adding a coupon to an order was updated to allow adding multiple coupons. See the video below.

1. Added new applied coupon list screen redirecting to the coupon details/edition screen.
In case there are no coupons in the order we redirect to coupon details directly to remove the redundant (coupon list) step.
2. Modified coupon addition and deletion logic in `OrderCreateEditViewModel` to support multiple coupon lines.

### Testing instructions

You'll need to create at least two coupons in wp-admin.

1. Create new order and try to add first coupon.
2. Verify you are redirected directly to coupon entry screen and that coupon is applied correctly to the order.
3. Try adding the second coupon by clicking coupons label and observe we show list of already applied coupons first. Click on the "+" FAB.
4. Enter the second coupon and verify both coupons are applied correctly to the order.

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/4527432/af7fe767-0992-486d-9cff-dd4cd0b890ea
